docs: Add more workflows to the GitHub Actions doc (#4146)
bentoml/BentoML: docs/source/guides/github-actions.rst

You can create a GitHub Action workflow to automate the process of building Bentos either from a GitHub repository or from a specified context path.

To use it, define a workflow file as below:

.. literalinclude:: ./../data/build-bento-action.yaml
   :language: yaml

This workflow first uses the `BentoML setup workflow <https://github.com/bentoml/setup-bentoml-action>`_ to prepare the necessary environment and
then builds the Bento using the default GitHub context. To specify a different context, set the ``context`` path:

.. code-block:: yaml

   - uses: bentoml/build-bento-action@v1
     with:
       context: 'path/to/context'

To specify a version, use the ``version`` field:

.. code-block:: yaml

   - uses: bentoml/build-bento-action@v1
     with:
       version: '1.0.0'

To specify a different ``bentofile.yaml`` file, use the ``bentofile`` field:

.. code-block:: yaml

   - uses: bentoml/build-bento-action@v1
     with:
       bentofile: 'path/to/bentofile'

.. note::

   ``bentofile`` must be a path relative to the ``context`` directory.

For more information, see the `build-bento-action repository <https://github.com/bentoml/build-bento-action>`_.

> Create and push Bento images to Docker Hub

Reviewer: ```suggestion
Create and push Bento images to Container registry
```

------------------------------------------

> You can create a GitHub Actions workflow to create and push Bento images to Docker Hub. This workflow uses `Docker Buildx <https://github.com/docker/buildx>`_,
> which offers extended capabilities for building containers, and is enhanced by the `Moby BuildKit builder toolkit <https://github.com/moby/buildkit>`_.

Reviewer: ```suggestion
You can create a GitHub Actions workflow to create and push Bento images toany container registry (Docker Hub, GHCR, ECR, GCR, etc.). This workflow uses `Docker Buildx <https://github.com/docker/buildx>`_,
which offers extended capabilities for building containers and is enhanced by the `Moby BuildKit builder toolkit <https://github.com/moby/buildkit>`_.
```

To use it, define a workflow file as below:

.. literalinclude:: ./../data/containerize-and-push.yaml
   :language: yaml

> This workflow first uses the `BentoML setup workflow <https://github.com/bentoml/setup-bentoml-action>`_ to prepare the necessary environment. It then
> sets up Docker QEMU and Docker Buildx, and configures Docker Hub authentication. Lastly, it builds a Bento, creates a Docker image, and pushes it to Docker Hub.

Reviewer: ```suggestion
In this example, the workflow first uses the `BentoML setup workflow <https://github.com/bentoml/setup-bentoml-action>`_ to prepare the necessary environment. It then
sets up Docker QEMU and Docker Buildx, and configures Docker Hub authentication. Lastly, it builds a Bento, creates a Docker image, and pushes it to Docker Hub.

.. note::

	You can also use ``docker/login-action@v2`` to login into other container registry that is supported by that action. Refer to https://github.com/docker/login-action?tab=readme-ov-file#usage for more information.
```
